4_swift_macros/StringifyMacro/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# This source file is part of the Swift open source project
+#
+# Copyright (c) 2024 Apple Inc. and the Swift project authors.
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+
+cmake_minimum_required(VERSION 3.22)
+if(POLICY CMP0157)
+  cmake_policy(SET CMP0157 NEW)
+endif()
+
+project(StringifyMacro Swift)
+
+if(NOT CMAKE_Swift_COMPILER_TARGET AND NOT SWIFT_MODULE_TRIPLE)
+  # SwiftSyntax requires that SWIFT_MODULE_TRIPLE is set or it won't emit the
+  # needed SwiftModule files.
+  execute_process(COMMAND ${CMAKE_Swift_COMPILER} -print-target-info
+    OUTPUT_VARIABLE SwiftTargetInfo)
+  string(JSON module_triple GET "${SwiftTargetInfo}" target moduleTriple)
+  set(SWIFT_MODULE_TRIPLE "${module_triple}" CACHE STRING "")
+  unset(module_triple)
+endif()
+
+# We use FetchContent to download and build SwiftSyntax. Unlike ExternalProject,
+# FetchContent does not re-invoke CMake, and the build graphs are merged. This
+# exposes the variables and build targets in SwiftSyntax to our Macro project.
+# Everything in this project is isolated, so you can have macros that require
+# different versions of SwiftSyntax. They won't be able to link eachother, but
+# will work together as imported by the compiler.
+include(FetchContent)
+FetchContent_Declare(SwiftSyntax
+  GIT_REPOSITORY https://github.com/apple/swift-syntax.git
+  GIT_TAG 247e3ce379141f81d56e067fff5ff13135bd5810)
```

Comment:
Mostly because pinning it will ensure that this example will continue building in the future. I would prefer basing it on a release tag, but I also just landed the needed change on `main`, so this is technically top-of-main at the moment.

Reply:
FWIW, pinning to an exact revision is also considered a security thing. It avoids a force push supply chain attack vector.